open_humans/templatetags/utilities.py

```diff
@@ -361,3 +364,15 @@ def make_badge(project, badge_class='oh-badge'):
 @register.simple_tag()
 def template_bool(item):
     return bool(item)
+
+
+@register.simple_tag(takes_context=True)
+def get_next_url(context):
+    next = context['request'].GET.get('next', None)
+    if next:
+        return next
```

Comment:
There's something complicated going on here when the URL target we originally wanted had parameters. If this happens, the `next` parameter needs to be encoded -- otherwise, everything after the first `&` in the original URL target is lost.

This does actually happen, potentially quite often, during an OAuth2 authorization for a project.

The answer seems likely to involve using `urllib.parse.encode` at some point in this process? I'm not sure if it should be here (maybe?) and/or if some additional decode also needs to be done on login.